x-pack/plugins/features/public/features_api_client.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { CoreSetup } from 'src/core/public';
+import { FeatureConfig, Feature } from '.';
+
+export class FeaturesAPIClient {
+  constructor(private readonly http: CoreSetup['http']) {}
```

Comment:
I had gotten in to the `CoreSetup['xxx']` habit because I was bitten by inconsistent types in the past. I remember cases where the service's exposed `Setup` interface differed from that `CoreSetup` returned. (e.g., `CoreSetup` might `Pick<>` a subset of properties). That's not the case here, and I'm sure the service APIs have stablized since then, but I at least wanted to explain my thinking. I'll update this to use `HttpSetup`.

Reply:
All these inconsistencies *should* have been fixed by now. Do not hesitate to ping us if you find any.